testsuite/MDAnalysisTests/coordinates/test_rdkit.py

```diff
@@ -65,14 +92,146 @@ def test_no_coordinates(self):
+                            dtype=object)
+        u.add_TopologyAttr('elements', elements)
+        return u
+
+    @pytest.fixture
+    def peptide(self):
+        u = mda.Universe(GRO)
+        elements = mda.topology.guessers.guess_types(u.atoms.names)
+        u.add_TopologyAttr('elements', elements)
+        return u.select_atoms("resid 2-12")
+
+    @pytest.mark.parametrize("smi", ["[H]", "C", "O", "[He]"])
+    def test_single_atom_mol(self, smi):
+        u = mda.Universe.from_smiles(smi, addHs=False,
+                                     generate_coordinates=False)
+        mol = u.atoms.convert_to("RDKIT")
+        assert mol.GetNumAtoms() == 1
+
+    @pytest.mark.parametrize("resname, n_atoms, n_fragments", [
+        ("PRO", 14, 1),
+        ("ILE", 38, 1),
+        ("ALA", 20, 2),
+        ("GLY", 21, 3),
+    ])
+    def test_mol_from_selection(self, peptide, resname, n_atoms, n_fragments):
+        mol = peptide.select_atoms("resname %s" % resname).convert_to("RDKIT")
+        assert n_atoms == mol.GetNumAtoms()
+        assert n_fragments == len(Chem.GetMolFrags(mol))
+
+    @pytest.mark.parametrize("sel_str, atom_index", [
+        ("resid 1", 0),
+        ("resname LYS and name NZ", 1),
+        ("resid 34 and altloc B", 2),
+    ])
+    def test_monomer_info(self, pdb, sel_str, atom_index):
+        rdmol = Chem.MolFromPDBFile(PDB_full)
+        sel = pdb.select_atoms(sel_str)
+        umol = sel.convert_to("RDKIT")
+        atom = umol.GetAtomWithIdx(atom_index)
+        mi = atom.GetMonomerInfo()
+
+        for mda_attr, rd_attr in RDATTRIBUTES.items():
+            rd_value = getattr(mi, "Get%s" % rd_attr)()
+            mda_value = getattr(sel, "%s" % mda_attr)[atom_index]
+            if mda_attr == "names":
+                rd_value = rd_value.strip()
+            assert rd_value == mda_value
+
+    def test_identical_topology_mol2(self, mol2):
+        """Check stereochemistry on atoms and bonds (but not yet)"""
+        rdmol = mol2_mol()
+        umol = mol2.atoms.convert_to("RDKIT")
+        assert rdmol.HasSubstructMatch(umol, useChirality=False)
+        assert umol.HasSubstructMatch(rdmol, useChirality=False)
+
+    def test_identical_topology(self):
+        rdmol = smiles_mol()
+        u = mda.Universe(rdmol)
+        umol = u.atoms.convert_to("RDKIT")
+        assert rdmol.HasSubstructMatch(umol) and umol.HasSubstructMatch(rdmol)
+
+    def test_raise_requires_elements(self):
+        u = mda.Universe(mol2_molecule)
+        with pytest.raises(AttributeError) as e:
+            u.atoms.convert_to("RDKIT")
+            assert ("`elements` attribute is required for the RDKitConverter"
+                    in str(e.value))
+
+    def test_warn_guess_bonds(self, pdb):
+        pdb.delete_bonds(pdb.bonds)
+        ag = pdb.select_atoms("resnum 101 and segid A")
+        pdb.delete_bonds(ag.bonds)
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
As above, I think you can also save yourself a little bit of code complexity by just using pytest.warns here.